tests/integration/dump_test.go

```diff
@@ -122,7 +122,163 @@ func Test_Dump_SkipConsumers(t *testing.T) {
 
 			expected, err := readFile(tc.expectedFile)
 			assert.NoError(t, err)
-			assert.Equal(t, output, expected)
+			assert.Equal(t, expected, output)
+		})
+	}
+}
+
+func Test_Dump_SkipConsumers_34x(t *testing.T) {
+	tests := []struct {
+		name          string
+		stateFile     string
+		expectedFile  string
+		skipConsumers bool
+	}{
+		{
+			name:          "dump with skip-consumers",
+			stateFile:     "testdata/dump/002-skip-consumers/kong34.yaml",
+			expectedFile:  "testdata/dump/002-skip-consumers/expected.yaml",
+			skipConsumers: true,
+		},
+		{
+			name:          "dump with no skip-consumers",
+			stateFile:     "testdata/dump/002-skip-consumers/kong34.yaml",
+			expectedFile:  "testdata/dump/002-skip-consumers/expected-no-skip-34.yaml",
+			skipConsumers: false,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			runWhen(t, "enterprise", ">=3.4.0 <3.5.0")
+			teardown := setup(t)
+			defer teardown(t)
+
+			assert.NoError(t, sync(tc.stateFile))
+
+			var (
+				output string
+				err    error
+			)
+			if tc.skipConsumers {
+				output, err = dump(
+					"--skip-consumers",
+					"-o", "-",
+				)
+			} else {
+				output, err = dump(
+					"-o", "-",
+				)
+			}
+			assert.NoError(t, err)
+
+			expected, err := readFile(tc.expectedFile)
+			assert.NoError(t, err)
+			assert.Equal(t, expected, output)
+		})
+	}
+}
+
+func Test_Dump_SkipConsumers_35x(t *testing.T) {
+	tests := []struct {
+		name          string
+		stateFile     string
+		expectedFile  string
+		skipConsumers bool
+	}{
+		{
+			name:          "dump with skip-consumers",
+			stateFile:     "testdata/dump/002-skip-consumers/kong34.yaml",
+			expectedFile:  "testdata/dump/002-skip-consumers/expected.yaml",
+			skipConsumers: true,
+		},
+		{
+			name:          "dump with no skip-consumers",
+			stateFile:     "testdata/dump/002-skip-consumers/kong34.yaml",
+			expectedFile:  "testdata/dump/002-skip-consumers/expected-no-skip-35.yaml",
+			skipConsumers: false,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			runWhen(t, "enterprise", ">=3.5.0")
+			teardown := setup(t)
+			defer teardown(t)
+
+			assert.NoError(t, sync(tc.stateFile))
+
+			var (
+				output string
+				err    error
+			)
+			if tc.skipConsumers {
+				output, err = dump(
+					"--skip-consumers",
+					"-o", "-",
+				)
+			} else {
+				output, err = dump(
+					"-o", "-",
+				)
+			}
+			assert.NoError(t, err)
+
+			expected, err := readFile(tc.expectedFile)
+			assert.NoError(t, err)
+			assert.Equal(t, expected, output)
+		})
+	}
+}
+
+func Test_Dump_SkipConsumers_Konnect(t *testing.T) {
+	// TODO: remove skip once Konnect support is enabled.
+	t.Skip()
```

Comment:
Skip accepts message arguments which are printed when the test is about to be run.

This is helpful when reading tests logs. Having said that I suggest to put the comment in the skip call and maybe even add a link to an issue (if such exists). WDYT?